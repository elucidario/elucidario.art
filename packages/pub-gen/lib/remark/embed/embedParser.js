import path from "path";
import { parseNodeValue, isPubGenNodeValue } from "../../utils.js";
import { getPaths, readFile } from "@elucidario/pkg-paths";
import { visit } from "unist-util-visit";
import { Console } from "@elucidario/pkg-console";

const embedParser = (treeOptions) => {
    const { publication, lang, assets, version, pkg, ...extra } = treeOptions;

    const console = new Console(pkg);

    // default docxProcessor to pandoc while we don't implement remark-docx.
    let docxProcessor = "pandoc";
    if (typeof extra.docxProcessor !== "undefined")
        docxProcessor = extra.docxProcessor;

    return (tree, file) => {
        visit(tree, "text", async (node) => {
            if (!isPubGenNodeValue(node.value)) return;

            const { action, filePath, fileOptions } = parseNodeValue(
                node.value,
            );

            if (action !== "embed") return;

            const embedData = readFile(
                path.resolve(
                    getPaths().publications,
                    publication,
                    "content",
                    lang,
                    filePath,
                ),
            ).value;

            node.type = "html";
            node.value = embedData;
        });
    };
};

export default embedParser;
