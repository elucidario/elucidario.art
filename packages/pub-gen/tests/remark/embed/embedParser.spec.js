import path from "path";

import embedParser from "../../../lib/remark/embed/embedParser";
import { readFile, getPaths } from "@elucidario/pkg-paths";

import testProcessor from "../../mocks/remarkProcessor";

describe("embedParser", () => {
    const embedContent = readFile(
        path.resolve(
            getPaths().publications,
            "publicacao-teste",
            "content",
            "pt-br",
            "internal",
            "body",
            "nodes",
            "embed.md",
        ),
    ).content;

    it("should return embedParser node", async () => {
        const embedParserNode = await testProcessor(embedContent, {
            plugin: embedParser,
            pluginOptions: {
                publication: "publicacao-teste",
                lang: "pt-BR",
            },
            stringifyOptions: {},
            publication: "publicacao-teste",
            lang: "pt-BR",
        });
        expect(embedParserNode).toMatchSnapshot();
    });
});
