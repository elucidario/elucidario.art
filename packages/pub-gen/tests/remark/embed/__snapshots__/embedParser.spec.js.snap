// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`embedParser should return embedParser node 1`] = `
VFile {
  "cwd": "C:\\Users\\55119\\Elucid√°rio.art\\elucidario\\packages\\pub-gen",
  "data": {},
  "history": [],
  "messages": [],
  "value": "**sei la mil coisas**
",
}
`;
